tdrs-frontend/src/index.js

```diff
@@ -24,8 +24,28 @@ axios.defaults.xsrfCookieName = 'csrftoken'
 axios.defaults.xsrfHeaderName = 'X-CSRFToken'
 axios.defaults.withCredentials = true
 
+function devLogin(devEmail) {
+  const BACKEND_URL = process.env.REACT_APP_BACKEND_URL
+  axios
+    .post(`${BACKEND_URL}/login/cypress`, {
+      username: devEmail,
+      token: 'local-cypress-token',
+    })
+    .then(function (response) {
+      console.log(response)
+    })
+    .catch(function (error) {
+      console.log(error)
+    })
+  store.dispatch({ type: 'SET_AUTH', payload: { devEmail } })
+  console.log(`dispatched SET_AUTH(${devEmail})`)
+}
+
 // call auth_check
 const store = configureStore()
+if (process.env.REACT_APP_DEVAUTH) {
```

Comment:
is it possible to add another condition here to prevent the `devLogin` process from happening if there's already an active session? i've noticed my non-devauth user sometimes getting replaced mid-session.

Reply:
to make the state selection easier, i'd suggest moving this logic to `App.js`, within the component. 

something like this, although the actual location to select in `auth = useSelector((state) => state.auth)` needs to be figured out to make the `!auth` condition work.
```js
  const dispatch = useDispatch()
  const auth = useSelector((state) => state.auth) // --> have to look in reducers for actual state path

  // call auth_check
  if (process.env.REACT_APP_DEVAUTH && !auth) {
    devLogin(dispatch, process.env.REACT_APP_DEVAUTH)
  }
  dispatch(fetchAuth())
```

the reason i'd like to move it is because the `App` component is already encapsulated by a redux `Provider` (in the render of `index.js`) meaning that `App` has access to the entire redux store through use of the `useDispatch` and `useSelector` hooks. Therefore you don't have to create a new store with `configureStore`

Reply:
Hi Jan,
I've been looking at this. There should be no problem with the !auth check. It should also be possible to move the code, but is that really an organizational improvement? Currently App.js isn't concerned with useDispatch or useSelector (and doesn't import them.) Also, the code implemented in the PR doesn't create a new store. The configureStore() invocation in the file is the original one (where the app initially creates the store.)

Reply:
you're right - i misread the diff, sorry about that. since you're building off a previous pattern, i'm ultimately good with this.

just to elaborate on my original point a little, though, i would argue the `App` component is the "right" place for the devauth logic (as well as the `fetchAuth` logic, probably) because in general i think it's a bad idea to have procedural code updates outside of a component. it's better if components are responsible for getting/setting state as it's more predictable when events will dispatch and when components will rerender, and i think it's more in-line with react/redux best practices
https://redux.js.org/style-guide/#use-the-react-redux-hooks-api
one area where i'd like to look more closely is the `getState()` call inside the conditional - when state is updated, a component utilizing `useSelector` will rerender automatically and any conditional logic will be re-performed. with `getState()` outside of the store-provided component, i'm not sure if this would normally have the intended effect.

because your implementation is working at the same time as the existing auth (reloading of `index.js` vs upon store update or component render), this is okay and it works. otherwise it may be prone to race conditions or other weird conflicting behavior. i think the frontend auth implementation could use a little work to make things clearer and more reliable, but your changes aren't negatively contributing to anything there.